codegen/module_system.go

```diff
@@ -92,10 +100,68 @@ func NewDefaultModuleSystem(h *PackageHelper) (*module.System, error) {
 		)
 	}
 
-	// TODO: Register endpoint module class and type generators
+	middlewareSpecs := map[string]*MiddlewareSpec{}
```

Comment:
Agreed, I like them to be explicitly specified as an endpoint's dependencies, so module system can pick them up just like clients. But they are slightly different because middleware config may need to provide input for middlewares to work.